diff --git a/Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp b/Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp
index b24480e8f539..3b0c5689307c 100644
--- a/Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp
+++ b/Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp
@@ -3635,6 +3635,7 @@ void webkit_web_view_reload(WebKitWebView* webView)
 {
     g_return_if_fail(WEBKIT_IS_WEB_VIEW(webView));
 
+    webkitWebViewUpdateSettings(webView);
     getPage(webView).reload({ });
 }
 
diff --git a/Source/WebKit/UIProcess/WebPageProxy.cpp b/Source/WebKit/UIProcess/WebPageProxy.cpp
index ab18bc32c482..fb85a78344d3 100644
--- a/Source/WebKit/UIProcess/WebPageProxy.cpp
+++ b/Source/WebKit/UIProcess/WebPageProxy.cpp
@@ -1198,9 +1198,6 @@ void WebPageProxy::notifyProcessPoolToPrewarm()
 
 void WebPageProxy::setPreferences(WebPreferences& preferences)
 {
-    if (&preferences == m_preferences.ptr())
-        return;
-
     protectedPreferences()->removePage(*this);
     m_preferences = preferences;
     protectedPreferences()->addPage(*this);
diff --git a/Source/WebKit/WebProcess/WebPage/WebPage.cpp b/Source/WebKit/WebProcess/WebPage/WebPage.cpp
index 87529320fa35..2a69d5449384 100644
--- a/Source/WebKit/WebProcess/WebPage/WebPage.cpp
+++ b/Source/WebKit/WebProcess/WebPage/WebPage.cpp
@@ -506,6 +506,8 @@
 #include <WebKitAdditions/WebPreferencesDefaultValuesAdditions.h>
 #endif
 
+#include <filesystem>
+
 namespace WebKit {
 using namespace JSC;
 using namespace WebCore;
@@ -4802,6 +4804,25 @@ void WebPage::updatePreferences(const WebPreferencesStore& store)
 
     updateSettingsGenerated(store, settings);
 
+    const char* lbseEnabledToggler = getenv("LAYER_BASED_SVG_ENGINE_TOGGLER");
+    if (lbseEnabledToggler != NULL) {
+        if (std::filesystem::exists(std::filesystem::path {lbseEnabledToggler})) {
+            fprintf(stderr, "LAYER_BASED_SVG_ENGINE: 1 (activated by the %s toggler)\n", lbseEnabledToggler);
+            settings.setLayerBasedSVGEngineEnabled(true);
+        } else {
+            fprintf(stderr, "LAYER_BASED_SVG_ENGINE: 0 ('%s' the toggler doesn't exist)\n", lbseEnabledToggler);
+            settings.setLayerBasedSVGEngineEnabled(false);
+       }
+    } else {
+        const char* lbseEnabled = getenv("LAYER_BASED_SVG_ENGINE");
+        if (lbseEnabled != NULL && strcmp(lbseEnabled, "1") == 0) {
+            fprintf(stderr, "LAYER_BASED_SVG_ENGINE: '%s'\n", lbseEnabled);
+            settings.setLayerBasedSVGEngineEnabled(true);
+        } else {
+            settings.setLayerBasedSVGEngineEnabled(false);
+        }
+    }
+
 #if !PLATFORM(GTK) && !PLATFORM(WIN) && !PLATFORM(PLAYSTATION) && !PLATFORM(WPE)
     if (!settings.acceleratedCompositingEnabled()) {
         WEBPAGE_RELEASE_LOG(Layers, "updatePreferences: acceleratedCompositingEnabled setting was false. WebKit cannot function in this mode; changing setting to true");
